src/extension/background-script/actions/nostr/nip44EncryptOrPrompt.ts

```diff
@@ -0,0 +1,72 @@
+import { USER_REJECTED_ERROR } from "~/common/constants";
+import nostr from "~/common/lib/nostr";
+import utils from "~/common/lib/utils";
+import { getHostFromSender } from "~/common/utils/helpers";
+import {
+  addPermissionFor,
+  hasPermissionFor,
+} from "~/extension/background-script/permissions";
+import state from "~/extension/background-script/state";
+import { MessageNip44EncryptGet, PermissionMethodNostr, Sender } from "~/types";
+
+const nip44EncryptOrPrompt = async (
+  message: MessageNip44EncryptGet,
+  sender: Sender
+) => {
+  const host = getHostFromSender(sender);
+  if (!host) return;
+
+  try {
+    const hasPermission = await hasPermissionFor(
+      PermissionMethodNostr["NOSTR_NIP44ENCRYPT"],
+      host
+    );
+
+    if (hasPermission) {
+      const response = (await state.getState().getNostr()).nip44Encrypt(
+        message.args.peer,
+        message.args.plaintext
+      );
+      return { data: response };
+    } else {
+      const promptResponse = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
+      }>({
+        ...message,
+        action: "public/nostr/confirmEncrypt",
```

Comment:
[`f3bb66d` (#2653)](https://github.com/getAlby/lightning-browser-extension/pull/2653/commits/f3bb66d4980def01df5f8cf78e5a218eb55e3fb1)

there were separate screens introduced but removed in recent commits. don't know the reason though cc @staab

Reply:
The new screens were added to accommodate the batching that was being introduced. Since we're using the same interface now, the screens would be identical. If you guys want to differentiate between nip44 and nip04 we could do that, but I don't think there's a strong case for it, it's all the same operation in terms of semantics; nip 44 should really be considered v2 of the encryption defined by nip 04.

Reply:
I think Rene agreed one seems fine for now.